qemu/tests/qemu_guest_agent.py

```diff
@@ -181,14 +189,31 @@ def gagent_verify(self, params, vm):
         self.gagent.verify_responsive()
         logging.info(self.gagent.cmd("guest-info"))
 
+    @error_context.context_aware
     def setup(self, test, params, env):
         BaseVirtTest.setup(self, test, params, env)
         start_vm = params["start_vm"]
         if (not self.vm) and (start_vm == "yes"):
             vm = self.env.get_vm(params["main_vm"])
             vm.verify_alive()
             self.vm = vm
-            self.setup_gagent_in_guest(params, self.vm)
+
+            error_context.context("Check qemu-ga status before install it", logging.info)
+            session = self._get_session(params, vm)
+            o_ins = session.cmd_output(params.get("gagent_pkg_check_cmd"))
```

Comment:
Can we check this by the return code of the command, so that we can use the same routine to perform installation and launch service?

Reply:
Excetp gagent_pkg_check_cmd of windows guest, the others use return code.As "wmic product 'QEMU guest agent' list brief" cmd always return zero whether the pks is installed or not.

And I also do the following modification as you suggested yesterday .
        1).seperate two subfunction of check_install and check_status for
        windows and linux guest.
        2).delete windows class setup() function.
        3).delete function setup_gagent_in_guest().
        4).in fstrim function, use setup() stead of setup_gagent_in_guest()

Could you have a check?
Thanks.